src/main/java/org/mastodon/mamut/launcher/MastodonLauncher.java

```diff
@@ -611,9 +612,8 @@ private void loadMastodonProject( final String projectPath )
 				if ( projectPath == null )
 				{
 					// We have to use the JFileChooser to open folders.
-					// String path = IteratorUtils.last(
-					// RecentProjectsPanel.recentProjects );
-					file = FileChooser.chooseFile( true, this, null, new ExtensionFileFilter( "mastodon" ), "Open Mastodon Project", FileChooser.DialogType.LOAD, SelectionMode.FILES_AND_DIRECTORIES );
+					String path = IteratorUtils.last( RecentProjectsPanel.recentProjects );
```

Comment:
You are perfectly right. I relied on the assumption that `RecentProjects.add()` would add the element at the end of list, as `.add()` method mostly do, but in fact in this case the element is added at the beginning in the `add()`-method.

I changed it accordingly.

Nevertheless, I am still proposing to keep a `IterableUtils`-class with the method `firstOrNull` that may be reused in other occasions. I hope, you do not think that this would be over-engineered.

Reply:
`IterableUtils` are beautiful but I think it's over-engineered. BTW there's also a class [IteratorUtils  in apache-commons](https://commons.apache.org/proper/commons-collections/apidocs/org/apache/commons/collections4/IterableUtils.html).\

Multiple files for drag & drop is fine, if it's simpler.